Bottom: 762d36377ac0a275bb2db568742f4312c8f4d4d9
Top:    4d9c8659250fcb5c5422bb815ec983a5932e44fd
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-01-07 01:00:07 +0000

draw_funcs.c: Don't draw pins/vias in layer for non GUI exporters

The non-gui paths render pins and vias with draw_ppv, so no need to
also render them for each layer.


---

diff --git a/src/draw_funcs.c b/src/draw_funcs.c
index 30e42ac..bd3df87 100644
--- a/src/draw_funcs.c
+++ b/src/draw_funcs.c
@@ -430,6 +430,10 @@ draw_layer (LayerType *layer, const BoxType *drawn_area, void *userdata)
   if (layer_num >= max_copper_layer)
     return;
 
+  /* Pins, pads and vias are handled in elsewhere by exporters, don't duplicate */
+  if (!gui->gui)
+    return;
+
   r_search (PCB->Data->pin_tree, drawn_area, NULL, pin_inlayer_callback, layer);
 
   /* draw element pads */
